src/circuit.rs

```diff
@@ -47,6 +47,9 @@ pub trait Circuit
 where
     Self: Sized,
 {
+    /// An identifier constant that you can use to disseminate between similar
```

Comment:
I'm not totally sure "disseminate" is the proper term here.

I would just add a simple comment that this is the circuit identifier.